**1)Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean):**

**а) среднее арифметическое,**

**b) среднее квадратичное отклонение,**

**с) смещенную и несмещенную оценки дисперсий для данной выборки.**

In [1]:
import numpy as np
from math import factorial

In [2]:
arr=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
arr

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

а) Расчитаем среднее арифметическое по формуле $M(X) = \frac{1}{n} \sum_{i=1}^{n} x_i$

In [3]:
def mean_value(array):
    return sum(array)/len(array)
print(f'Среднее арифметическое для данной выборки М(Х) = {mean_value(arr): .2f}')

Среднее арифметическое для данной выборки М(Х) =  65.30


б) Расчитаем среднее квадратичное отклонение выборки

In [4]:
def mean_square_deviation(array):
    square_dev=(array-mean_value(array))**2
    return (sum(square_dev)/len(square_dev))**(1/2)

In [6]:
print(f'Среднее квадратичное отклонение для данной выборки = {mean_square_deviation(arr): .4f}')

Среднее квадратичное отклонение для данной выборки =  30.8239


в) Расчитаем смещенную и несмещенную оценку дисперсии

In [8]:
def dispers(array, no_off=False):
    square_dev=(array-mean_value(array))**2
    return sum(square_dev)/(len(square_dev)-1) if no_off else sum(square_dev)/len(square_dev)
print(f'Смещенная оценка дисперсии для данной выборки = {dispers(arr): .4f}\n'
      f'Немещенная оценка дисперсии для данной выборки = {dispers(arr, True): .4f}'
     )

Смещенная оценка дисперсии для данной выборки =  950.1100
Немещенная оценка дисперсии для данной выборки =  1000.1158


**Ответ**

а) $М(Х) = 65.30$

b) $σ = 30.8239$

c)$\sigma_{смещ}^2$ = $950.1100$,  $\sigma_{несмещ}^2$ = $1000.1158$

**2)В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?**

Пусть А
-искомое событие "из двух корзин вытащили 3 белых" мяча. Тогда это событие может произойти следующими способами:

а) А(1) из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча;

б) А(2) из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча;

в) А(3) из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч.

Вероятность события А будет определяться как сумма вероятностей этих событий.

Определим вероятность события А(1):

In [9]:
def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

In [10]:
P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча'
      f', из второй вытащили 3 белых мяча"\nP(A1) = {P1: .4f}'
     )

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(A1) =  0.0152


Определим вероятность события А(2):

In [11]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч'
      f', из второй вытащили 2 белых мяча"\nP(A2) = {P2: .4f}'
     )

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(A2) =  0.2273


Определим вероятность события А(3):

In [12]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча'
      f', из второй вытащили 1 белый мяч"\nP(A2) = {P3: .4f}'
     )

Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(A2) =  0.1263


Вычислим $Р(А)$:

In [13]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


**Ответ**

Вероятность того, что из вытащенных мячей 3 белые $Р(А)$ = $36.87$%

**3)На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6.**

**Найти вероятность того, что выстрел произведен:**

**a) первым спортсменом;**

**б) вторым спортсменом;**

**в) третьим спортсменом.**

In [14]:
PB=1/3
PA=PB*0.9+PB*0.8+PB*0.6
print(f'Полная вероятность наступления события А Р(А) = {PA: .4f}')

Полная вероятность наступления события А Р(А) =  0.7667


Теперь, исходя из того, что событие А уже произошло, с помощью формулы Байеса оценим вероятность того, что имело место одно из событий

In [15]:
PAB1=PB*0.9/PA
PAB2=PB*0.8/PA
PAB3=PB*0.6/PA
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {PAB1: .4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {PAB2: .4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {PAB3: .4f}.'
     )

Вероятность того, что выстрел произвёл первый спортсмен:  0.3913;
Вероятность того, что выстрел произвёл второй спортсмен:  0.3478;
Вероятность того, что выстрел произвёл третий спортсмен:  0.2609.


**Ответ**

а) Вероятность того, что выстрел произвёл первый спортсмен: $39.13$%

б) Вероятность того, что выстрел произвёл второй спортсмен: $34.78$%

в) Вероятность того, что выстрел произвёл третий спортсмен: $26.09$%

**4)В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9.**

**Студент сдал первую сессию. Какова вероятность, что он учится:**

**a) на факультете A;**

**б) на факультете B;**

**в) на факультете C?**

Пусть $n$ - количество студентов на факультете $A$, тогда на факультете $B$ учится $n$ студентов, а на факультете $C$ $2n$ студентов.
Введем события S(A), S(B), S(C), что сессию сдаёт студент факультетов $A$, $B$, $C$, соответственно, и событие $D$ - сессия сдана. Cобытия S(A), S(B), S(C)образуют полную группу событий, P(S(A))+P(S(B))+P(S(C)) = 1. Т.е. 1/n + 1/n + 1/2n = 1n = 2.5. P(S(A)) = 0,25; P(S(В)) = 0,25; P(S(С)) = 0,5

In [16]:
PD=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность наступления события D, P(D) = {PD}.')

Полная вероятность наступления события D, P(D) = 0.825.


По формуле Байеса вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию.

In [17]:
PDSA=0.25*0.8/PD
PDSB=0.25*0.7/PD
PDSC=0.5*0.9/PD
print(f'Вероятность того, что студент учится на факультете А: {PDSA: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {PDSB: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {PDSC: .4f}.'
      )

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


**Ответ**

Вероятность того, что студент учится на факультете А: $24.24$%

Вероятность того, что студент учится на факультете B: $21.21$%

Вероятность того, что студент учится на факультете C: $55.55$%

**5) Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:**

**а) все детали;**

**б) только две детали;**

**в) хотя бы одна деталь;**

**г) от одной до двух деталей?**

Введем следующие события:
а) Событие ${A_1}$ - вышла из строя первая деталь, по условию задачи - вероятность этого события: $P$(${A_1})$ = $0.1$;

б) Событие $\overline{A_1}$ - противоположное событию ${A_1}$ - первая деталь не вышла из строя, вероятность этого события: $P$($\overline{A_1})$ = $1$ - $P$(${A_1})$ = $1 - 0.1 = 0.9$;

в) Событие ${A_2}$ - вышла из строя вторая деталь, по условию задачи - вероятность этого события: $P$(${A_1})$ = $0.2$;

г) Событие $\overline{A_2}$ - противоположное событию ${A_2}$ - вторая деталь не вышла из строя, вероятность этого события: $P$($\overline{A_2})$ = $1$ - $P$(${A_2})$ = $1 - 0.2 = 0.8$;

д) Событие ${A_3}$ - вышла из строя первая деталь, по условию задачи - вероятность этого события: $P$(${A_3})$ = $0.25$;

е) Событие $\overline{A_3}$ - противоположное событию ${A_3}$ - третья деталь не вышла из строя, вероятность этого события: $P$($\overline{A_3})$ = $1$ - $P$(${A_3})$ = $1 - 0.25 = 0.9$;

а) Для того, чтобы произошло событие "вышли из строя все детали", необходимо, чтобы одновременно произошли события ${A_1}$, ${A_2}$, ${A_3}$. А вероятностью этого события $P(3)$ будет произведением вероятностей этих событий.

In [18]:
P3=0.1*0.2*0.25
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P3: .4f}')

Вероятность того, что из строя выйдут все детали Р(3) =  0.0050


б) Для того, чтобы произошло событие "вышли из строя 2 детали", необходимо что бы случилось одна из следующих комбинаций событий:

Произошли события  ${A_1}$ и  ${A_1}$ и $\overline{A_3}$;

Произошли события  ${A_1}$ и  ${A_3}$ и $\overline{A_2}$;

Произошли события ${A_2}$ и  ${A_3}$ и $\overline{A_1}$.

In [19]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что из строя выйдут только 2 детали Р(2) = {P2: .4f}')

Вероятность того, что из строя выйдут только 2 детали Р(2) =  0.0800


в) Событие "выйдет из строя хотя бы одна деталь" можно рассмотреть как противоположное событию "не вышло из строя ни одной детали".

Пусть $P(0)$ вероятность события "не вышло из строя ни одной детали". Данное событие может произойти при одновременном наступлении событий $\overline{A_1}$ и $\overline{A_2}$ и $\overline{A_3}$


In [20]:
P0=0.9*0.8*0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P0: .4f}')

Вероятность того, что из строя не выйдет ни одной детали Р(0) =  0.5400


Тогда вероятность события "выйдет из строя хотя бы одна деталь:

In [21]:
P_0=1-P0
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = {P_0: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) =  0.4600


г) Событие "из строя выйдут от одной до двух деталей" произойдет при наступлении события "из строя выйдет одна деталь" или из строя выйдут 2 детали, следовательно его вероятность будет равна сумме вероятностей этих событий

In [23]:
P1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'Вероятность того, что выйдет из строя одна деталь Р(1) = {P1: .4f}')

Вероятность того, что выйдет из строя одна деталь Р(1) =  0.3750


In [26]:
P12=0.375+0.08
print(f'Вероятность того, что из строя выйдут от одной до двух деталей Р(12) = {P12: .4f}')

Вероятность того, что выйдет из строя выйдут от одной до двух деталей Р(12) =  0.4550


**Ответ**

Вероятность того, что из строя выйдут все детали = $0.50$%

Вероятность того, что из строя выйдут только 2 детали = $0.8$%

Вероятность того, что выйдет из строя хотя бы одна деталь = $46$%

Вероятность того, что выйдет из строя выйдут от одной до двух деталей = $45.5$%